src/ddsrt/include/dds/ddsrt/events/darwin.h

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright(c) 2006 to 2020 ADLINK Technology Limited and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+ * v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+#ifndef DDSRT_EVENTS_DARWIN_H
+#define DDSRT_EVENTS_DARWIN_H
+
+#if defined (__cplusplus)
+extern "C" {
+#endif
+
+#if defined (__cplusplus)
+}
+#endif
+#endif /* DDSRT_EVENTS_DARWIN_H */
```

Comment:
Github is unhappy with files not ending on a newline, and it is not the only one:

https://stackoverflow.com/questions/8172246/what-is-the-meaning-of-warning-no-new-line-at-end-of-file?lq=1
> The C language requires that every source file must end with a newline (from C99 5.1. 1.2/1): A source file that is not empty shall end in a new-line character, which shall not be immediately preceded by a backslash character before any such splicing takes place.

I am not aware of any compilers, other than long-obsolete version of some IDL compiler, that uses the freedom such a rule gives. But as a general rule, it makes sense to have the files end with a newline. You can probably even configure your editor to make sure of it :)

Reply:
I already saw git complaining about this, though I did not know that this was part of the C language standard.
I will change the files to end on a newline.

Reply:
This is now implemented.